### DISCUSSION TITLE
When using UVICORN to open multiple workers for concurrent testing under Windows, the program freezes without any error or exception messages output

### DISCUSSION THREAD

**Discussion by Yasuo369** [fastapi/fastapi#11665] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
import uvicorn

app = FastAPI()


@app.get('/say')
async def my_hello():
    return {'message': 'hello!'}

if __name__ == '__main__':
    uvicorn.run(app=app, host='0.0.0.0', port=8001, workers=2)
```

### Description

I opened two workers to run the program using UVICORN in Windows, and used JMeter for concurrent interface testing. There were random cases where the program did not respond, FASTAPI did not respond to requests properly, and JMeter did not issue requests. When I randomly requested an interface, the program returned to normal

### Operating System

Windows

### Operating System Details

windows 10 and windows 11

### FastAPI Version

 0.111.0

### Pydantic Version

1.10.4

### Python Version

3.8.3

**Comment by phatneglo** [comment]
You're having issues with FastAPI freezing when using multiple Uvicorn workers on Windows with JMeter testing. No errors show up, so it's hard to debug.

Try this:

1. Add logging by putting this at the top of your code:

```python
import logging
logging.basicConfig(level=logging.DEBUG)
```

This will log everything, so you may see clues in the logs.

2. If logging doesn't help, disable workers by changing:

```python
uvicorn.run(app=app, host='0.0.0.0', port=8001, workers=2)
```

to 

```python
uvicorn.run(app=app, host='0.0.0.0', port=8001, workers=1)
```

3. If it still freezes, check your code for potential deadlocks or race conditions.

4. As a last resort, try a different testing tool like Locust instead of JMeter.

If none of that works, try to upgrade your python looks like old version. cheers!

**Reply by Yasuo369** [reply]
Thank you very much for your help, but this problem currently only appears when workers are greater than 1, and only appears on Windows, not under Linux. I've tried to replace the new version of python and fastapi and still have this issue.

**Comment by mskrip** [comment]
Not sure if related, but if you are setting `workers` parameter, the `app` needs to be provided as import string, so

```python
uvicorn.run(app="script:app", host="0.0.0.0", port=8001, workers=2)
```

**Reply by Yasuo369** [reply]
I don't quite understand what you mean, can you elaborate on that
